privacy-contracts/src/main/solidity/DefaultOnChainPrivacyGroupManagementContract.sol

```diff
@@ -12,54 +13,58 @@ contract DefaultOnChainPrivacyGroupManagementContract is OnChainPrivacyGroupMana
         return _version;
     }
 
-    // overrides
     function canExecute() external view returns (bool) {
         return _canExecute;
     }
 
     function lock() public {
         require(_canExecute);
+        require(tx.origin == _owner, "Origin not the owner.");
```

Comment:
The owner check can be simplified using a function modifier. Check this example:
https://solidity.readthedocs.io/en/v0.6.2/contracts.html#function-modifiers

Reply:
Let's talk about that ...